File: bin/src/poc/implementation/search/test_data/sample.go
// Go sample
package main

import "fmt"

type Person struct {
    Name string
    Age  int
}

func (p *Person) Greet() {
    fmt.Printf("Hello, I'm %s\n", p.Name)
}

func NewPerson(name string, age int) *Person {
    return &Person{Name: name, Age: age}
}

func main() {
    person := NewPerson("Bob", 25)
    person.Greet()
}

File: bin/src/poc/implementation/search/test_data/sample.rb
# Ruby sample
class Person
  attr_reader :name, :age
  
  def initialize(name, age)
    @name = name
    @age = age
  end
  
  def greet
    puts "Hello, I'm #{@name}"
  end
end

def create_person(name, age)
  Person.new(name, age)
end

# Main
person = create_person("Charlie", 35)
person.greet